### PR TITLE
[daint-gpu] Fix a typo in the intel and pgi compilers version

### PR DESCRIPTION
**Pull request by victorusu**

### CODE REVIEW

File: spack/daint/compilers.yaml [hunk 1]
```diff
@@ -63,6 +63,8 @@ compilers:
     modules:
     - PrgEnv-intel
     - intel/19.0.1.144
+    - cdt/20.08
+    - intel/19.0.1.144
```
Comment (ekouts) [inline spack/daint/compilers.yaml:67]: You have added this twice

File: spack/daint/compilers.yaml [hunk 2]
```diff
@@ -78,6 +80,8 @@ compilers:
     modules:
     - PrgEnv-intel
     - intel/19.1.1.217
+    - cdt/20.08
+    - intel/19.1.1.217
```
Comment (ekouts) [inline spack/daint/compilers.yaml:84]: same here

File: spack/daint/compilers.yaml [hunk 3]
```diff
@@ -93,6 +97,8 @@ compilers:
     modules:
     - PrgEnv-pgi
     - pgi/20.1.0
+    - cdt/20.08
+    - pgi/20.1.0
```
Comment (ekouts) [inline spack/daint/compilers.yaml:101]: same

File: spack/daint/compilers.yaml [hunk 4]
```diff
@@ -108,6 +114,8 @@ compilers:
     modules:
     - PrgEnv-pgi
     - pgi/20.1.1
+    - cdt/20.08
+    - pgi/20.1.1
```
Comment (ekouts) [inline spack/daint/compilers.yaml:118]: same

**Review by ekouts — CHANGES_REQUESTED** [review]

File: spack/daint/compilers.yaml (hunk 1 above)
Comment (victorusu) [reply spack/daint/compilers.yaml:67]: Yes! Due to Spack's limitation, you cannot have the `cdt` defined before the first `intel` entry. And whenever we load the `cdt` module the compiler is reset. So we need to load again the `intel` module.

**Review by ekouts — APPROVED** [review]

**Pull request merged**